src/lib.rs

```diff
@@ -257,23 +255,24 @@ impl<K: EnrKey> Enr<K> {
         self.seq
     }
 
+    /// # Panics
+    ///
+    /// Will panic if data is not sanitized
     /// Reads a custom key from the record if it exists, decoded as data.
     #[allow(clippy::missing_panics_doc)]
-    pub fn get(&self, key: impl AsRef<[u8]>) -> Option<&[u8]> {
+    pub fn get(&self, key: impl AsRef<[u8]>) -> Option<impl AsRef<[u8]>> {
```

Comment:
This removed the vec, but since `insert_raw_rlp` already receives `Bytes`, the type is already part of our public API. Let's just return `Bytes` here instead of the `impl AsRef<[u8]>`

This should also allow you to change things like 
```diff
- assert_eq!(decoded_enr.get("custom_key").as_ref().map(AsRef::as_ref), Some(vec![0,0,1]).as_deref());
+ assert_eq!(decoded_enr.get("custom_key").unwrap(), &[0,0,1]);
```
which is way cleaner